src/endf.cpp

```diff
@@ -0,0 +1,261 @@
+#include "openmc/endf.h"
+
+#include <algorithm> // for copy
+#include <array>
+#include <cmath>     // for log, exp
+#include <iterator>  // for back_inserter
+#include <stdexcept> // for runtime_error
+
+#include "xtensor/xarray.hpp"
+#include "xtensor/xview.hpp"
+
+#include "openmc/constants.h"
+#include "openmc/hdf5_interface.h"
+#include "openmc/search.h"
+
+namespace openmc {
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+Interpolation int2interp(int i)
+{
+  // TODO: We are ignoring specification of two-dimensional interpolation
+  // schemes (method of corresponding points and unit base interpolation). Those
+  // should be accounted for in the distribution classes somehow.
+
+  switch (i) {
+  case 1: case 11: case 21:
+    return Interpolation::histogram;
+  case 2: case 12: case 22:
+    return Interpolation::lin_lin;
+  case 3: case 13: case 23:
+    return Interpolation::lin_log;
+  case 4: case 14: case 24:
+    return Interpolation::log_lin;
+  case 5: case 15: case 25:
+    return Interpolation::log_log;
+  default:
+    throw std::runtime_error{"Invalid interpolation code."};
+  }
+}
+
+bool is_fission(int mt)
+{
+  return mt == 18 || mt == 19 || mt == 20 || mt == 21 || mt == 38;
+}
+
+bool is_disappearance(int mt)
+{
+  if (mt >= N_DISAPPEAR && mt <= N_DA) {
+    return true;
+  } else if (mt >= N_P0 && mt <= N_AC) {
+    return true;
+  } else if (mt == N_TA || mt == N_DT || mt == N_P3HE || mt == N_D3HE
+    || mt == N_3HEA || mt == N_3P) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+bool is_inelastic_scatter(int mt)
+{
+  if (mt < 100) {
```

Comment:
Optional: For `100` here, and for `27` on line 69 and `200` on line 71, might make it more readable to define them as constants (similar to `N_2N0`) or leave a comment clarifying their meaning.